packages/statemanager/src/statelessVerkleStateManager.ts

```diff
@@ -0,0 +1,634 @@
+  zeros,
+} from '@ethereumjs/util'
+import { getKey, getStem, verifyUpdate } from '@ethereumjs/verkle'
+import debugDefault from 'debug'
+import { keccak256 } from 'ethereum-cryptography/keccak.js'
+import { concatBytes, equalsBytes } from 'ethereum-cryptography/utils'
+
+import { AccountCache, CacheType, StorageCache } from './cache/index.js'
+import { OriginalStorageCache } from './cache/originalStorageCache.js'
+
+import type { DefaultStateManager } from './stateManager.js'
+import type { VerkleExecutionWitness } from '@ethereumjs/block'
+import type {
+  AccountFields,
+  Common,
+  EVMStateManagerInterface,
+  Proof,
+  StorageDump,
+  StorageRange,
+} from '@ethereumjs/common'
+import type { Address, PrefixedHexString } from '@ethereumjs/util'
+
+const { debug: createDebugLogger } = debugDefault
+
+const debug = createDebugLogger('statemanager:verkle')
+
+export interface VerkleState {
+  [key: PrefixedHexString]: PrefixedHexString
+}
+
+export interface EncodedVerkleProof {
+  [key: PrefixedHexString]: PrefixedHexString
+}
+
+type CacheOptions = {
+  /**
+   * Allows for cache deactivation
+   *
+   * Depending on the use case and underlying datastore (and eventual concurrent cache
+   * mechanisms there), usage with or without cache can be faster
+   *
+   * Default: false
+   */
+  deactivate?: boolean
+
+  /**
+   * Cache type to use.
+   *
+   * Available options:
+   *
+   * ORDERED_MAP: Cache with no fixed upper bound and dynamic allocation,
+   * use for dynamic setups like testing or similar.
+   *
+   * LRU: LRU cache with pre-allocation of memory and a fixed size.
+   * Use for larger and more persistent caches.
+   */
+  type?: CacheType
+
+  /**
+   * Size of the cache (only for LRU cache)
+   *
+   * Default: 100000 (account cache) / 20000 (storage cache)
+   *
+   * Note: the cache/trie interplay mechanism is designed in a way that
+   * the theoretical number of max modified accounts between two flush operations
+   * should be smaller than the cache size, otherwise the cache will "forget" the
+   * old modifications resulting in an incomplete set of trie-flushed accounts.
+   */
+  size?: number
+}
+
+type CacheSettings = {
+  deactivate: boolean
+  type: CacheType
+  size: number
+}
+
+/**
+ * Options dictionary.
+ */
+export interface StatelessVerkleStateManagerOpts {
+  accountCacheOpts?: CacheOptions
+  /**
+   * The common to use
+   */
+  common?: Common
+  storageCacheOpts?: CacheOptions
+}
+
+/**
+ * Tree key constants.
+ */
+const VERSION_LEAF_KEY = toBytes(0)
+const BALANCE_LEAF_KEY = toBytes(1)
+const NONCE_LEAF_KEY = toBytes(2)
+const CODE_KECCAK_LEAF_KEY = toBytes(3)
+const CODE_SIZE_LEAF_KEY = toBytes(4)
+
+const HEADER_STORAGE_OFFSET = 64
+const CODE_OFFSET = 128
+const VERKLE_NODE_WIDTH = 256
+const MAIN_STORAGE_OFFSET = 256 ** 31
+
+const PUSH_OFFSET = 95
+const PUSH1 = PUSH_OFFSET + 1
+const PUSH32 = PUSH_OFFSET + 32
+
+export class StatelessVerkleStateManager implements EVMStateManagerInterface {
```

Comment:
Please add code docs here with a class description/overview and an `experimental` note.

Reply:
Done